src/GafferSceneUI/VisualiserTool.cpp

```diff
@@ -1238,6 +1286,46 @@ class VisualiserGadget : public Gadget
 										data->writable() = iData->readable()[i];
 										vertexValue = data;
 									}
+									if( auto fData = runTimeCast<const FloatVectorData>( vData.get() ) )
+									{
+										auto data = runTimeCast<FloatData>( vertexValue );
+										if( !data )
+										{
+											data.reset( new FloatData() );
+										}
+										data->writable() = fData->readable()[i];
+										vertexValue = data;
+									}
+									if( auto v2fData = runTimeCast<const V2fVectorData>( vData.get() ) )
+									{
+										auto data = runTimeCast<V2fData>( vertexValue );
+										if( !data )
+										{
+											data.reset( new V2fData() );
+										}
+										data->writable() = v2fData->readable()[i];
+										vertexValue = data;
+									}
+									if( auto v3fData = runTimeCast<const V3fVectorData>( vData.get() ) )
+									{
+										auto data = runTimeCast<V3fData>( vertexValue );
+										if( !data )
+										{
+											data.reset( new V3fData() );
+										}
+										data->writable() = v3fData->readable()[i];
+										vertexValue = data;
+									}
+									if( auto c3fData = runTimeCast<const Color3fVectorData>( vData.get() ) )
+									{
+										auto data = runTimeCast<Color3fData>( vertexValue );
+										if( !data )
+										{
+											data.reset( new Color3fData() );
+										}
+										data->writable() = c3fData->readable()[i];
+										vertexValue = data;
+									}
```

Comment:
Why are we going to all this trouble of making an intermediate per-vertex Data value? Why don't we just format to a string directly here?

Reply:
The original motivation was to use the same data for the drag value. That's not obsolete in favor of https://github.com/GafferHQ/gaffer/commit/0b2404204c466b409ae368692ac20579653fb54b